marshal/base-types.c

```diff
@@ -75,26 +71,36 @@ type##_Marshal ( \
          (uintptr_t)&src, \
          (uintptr_t)buffer, \
          local_offset); \
-    src = marshal_func (src); \
-    memcpy (&buffer [local_offset], &src, sizeof (src)); \
+\
+    switch (sizeof(type)) { \
+        case 1: \
+            break; \
+        case 2: \
+            src = HOST_TO_BE_16(src); \
+            break; \
+        case 4: \
+            src = HOST_TO_BE_32(src); \
+            break; \
+        case 8: \
+            src = HOST_TO_BE_64(src); \
+            break; \
+\
+    } \
+    memcpy (&buffer [local_offset], &src, sizeof(src)); \
     if (offset != NULL) { \
-        *offset = local_offset + sizeof (src); \
+        *offset = local_offset + sizeof(src); \
```

Comment:
This is another style thing: I typically include a space after the function name and before the opening paren / `(`. I do the same for built-ins like `sizeof`. Is this unusual? Whatever the result we should document this in the CONTRIBUTING.md.